Create Term Entry for Python String function `.endswith()`

Pandz18 opened the pull request:
### Description

Created an entry for the in-built python function, `.endswith()`.

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

<!-- Please check ALL the boxes: -->

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

Pandz18 commented:
Also a question, I had linked my GitHub profile to my codecademy account but for the reset submission, it still shows as anonymous contributor.

KTom101 commented on content/python/concepts/built-in-functions/terms/endswith/endswith.md:
```diff
@@ -0,0 +1,75 @@
+check = example.endswith("st")
+print(check)
+
+# Output: False
```
Changing to reflect added index values.
```suggestion
# Output: True
```

KTom101 commented on content/python/concepts/built-in-functions/terms/endswith/endswith.md:
````diff
@@ -0,0 +1,75 @@
+```
+
+## Example 1
+
````
```suggestion

Checking for the value `g` at the end of the string, `example`:

```

KTom101 commented on content/python/concepts/built-in-functions/terms/endswith/endswith.md:
````diff
@@ -0,0 +1,75 @@
+```
+
+## Example 2
+
````
```suggestion

Checking for the value `s`:

```

KTom101 commented on content/python/concepts/built-in-functions/terms/endswith/endswith.md:
````diff
@@ -0,0 +1,75 @@
+```
+
+## Example 4
+
````
Let's change this example to have a starting and ending index.
```suggestion

The following checks for `st` and has a starting index of `5` and an ending index of `12`:

```

KTom101 commented on content/python/concepts/built-in-functions/terms/endswith/endswith.md:
````diff
@@ -0,0 +1,75 @@
+```
+
+## Example 3
+
````
```suggestion

Checking for the value `ring`:
```

Pandz18 replied:
@KTom101 Hey, just updated the file, hope it's fine now!

KTom101 reviewed (commented):
Hello @Pandz18, I've completed my review for this entry.  Please take a look at these changes and let us know when you're done making revisions so we can move this on to the 2nd review. Thanks!

KTom101 reviewed (approved):
@Pandz18 Great! This is ready for a second review! 
@Dusch4593 @yangc95

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
```diff
@@ -0,0 +1,89 @@
+---
+Title: 'endswith()'
```
```suggestion
Title: '.endswith()'
```

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
```diff
@@ -0,0 +1,89 @@
+  - 'paths/data-science'
+---
+
+The `endswith()` string method checks to see if a particular value is attached to the end of the string it is called on and returns a boolean value.
```
```suggestion
The `.endswith()` method returns `True` if a particular value is attached to the end of a given string. Otherwise, it returns `False`.
```

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
```diff
@@ -0,0 +1,89 @@
+The `endswith()` string method checks to see if a particular value is attached to the end of the string it is called on and returns a boolean value.
+
+## Syntax
+
```
```suggestion

With a given `string`, the `.endswith()` method can be used in the following way:

```

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
````diff
@@ -0,0 +1,89 @@
+string.endswith(value, start, end)
+```
+
+## Parameters
+
+- `value` (required): value to be checked at the end of the string
+- `start` (optional): specifies the starting index
+- `end` (optional): specifies the ending index
````
```suggestion
The `value` is required argument that is checked for whether it exists at the beginning of the `string`. 

Conversely, the `value` can be tested whether it is the end of a particular substring of `string` by using the optional `start` and `end` index arguments.
```

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
````diff
@@ -0,0 +1,89 @@
+# Output: True
+```
+
+**Note:** The `endswith()` function is case-sensitive and is demonstrated by the example given below.
+
+## Example 5
````
```suggestion
## Codebyte example
```

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
````diff
@@ -0,0 +1,89 @@
+
+## Example 5
+
+```py
````
```suggestion
```codebyte/py
```

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
````diff
@@ -0,0 +1,89 @@
+## Example 5
+
+```py
+example = "This is a string"
+check = example.endswith("G")
+print(check)
+
+# Output: False
````
```suggestion
example_A = "This is a string"
check_A = example_A.endswith("ring")
print(check_A)

// .endswith() is case-sensitive
example_B = "This is a string"
check_B = example_B.endswith("G")
print(check_B)
```

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
````diff
@@ -0,0 +1,89 @@
+# Output: False
+```
+
+## Example 3
+
+Checking for the value `ring`:
+
+```py
+example = "This is a string"
+check = example.endswith("ring")
+print(check)
+
+# Output: True
+```
````
I think we should go with just three non-codebyte example. Could we delete lines #55-65 here?

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
````diff
@@ -0,0 +1,89 @@
+# Output: True
+```
+
+## Example 4
````
```suggestion
```

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
````diff
@@ -0,0 +1,89 @@
+# Output: False
+```
+
+## Example 3
+
+Checking for the value `ring`:
````
```suggestion
The next example returns `True` after checking for the value `ring`:
```

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
```diff
@@ -0,0 +1,89 @@
+- `start` (optional): specifies the starting index
+- `end` (optional): specifies the ending index
+
+## Example 1
+
+Checking for the value `g` at the end of the string, `example`:
```
```suggestion
## Examples

The following ckecks for the value `g` at the end of the `example` string:
```

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
````diff
@@ -0,0 +1,89 @@
+# Output: True
+```
+
+## Example 2
+
+Checking for the value `s`:
````
```suggestion
This example returns `False` after checking for the value `s`:
```

Dusch4593 reviewed (changes requested):
Hey @Pandz18! Thank you for this latest entry! In c71a57aa81803f4ff82e622c4e5a0915b807797d, I moved the term folder + file from the `python/concepts/built-in-functions/terms` directory to `python/concepts/strings/terms` directory because `.endswith()` is technically a String method.

But other than that, this entry is off to a solid start! 💪 I went ahead and left comments/suggestions for you to look over. After you're finish pushing the next batch of changes, tag me so we can proceed onward. 😃

Pandz18 commented:
Hey, @Dusch4593, sorry for the delay, so here's the update for the readme:
- Deleted the extra example and removed all the headings example under a common one , called examples.
- made changes to the wording.

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
````diff
@@ -0,0 +1,72 @@
+# Output: True
+```
+
+## Codebyte example
````
```suggestion
## Codebyte Example
```

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
````diff
@@ -0,0 +1,72 @@
+
+## Codebyte example
+
+```codebyte/py
````
```suggestion
```codebyte/python
```

Dusch4593 replied:
For regular snippets, let's use \```py and for Codebytes let's use \```codebyte/python

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
```diff
@@ -0,0 +1,72 @@
+---
+Title: '.endswith()'
+Description: 'Checks whether or not a string ends with the given value.'
```
```suggestion
Description: 'Checks whether or not a string ends with a given value.'
```

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
```diff
@@ -0,0 +1,72 @@
+  - 'paths/data-science'
+---
+
+TThe `.endswith()` method returns `True` if a particular value is attached to the end of a given string. Otherwise, it returns `False`.
```
```suggestion
The `.endswith()` method checks a value against a given string and returns `True` if the value exists at the end of that string. Otherwise, it returns `False`.
```

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
```diff
@@ -0,0 +1,72 @@
+check_A = example_A.endswith("ring")
+print(check_A)
+
+// .endswith() is case-sensitive
```
```suggestion
```

Let's avoid using comments in Codebytes whenever possible (this one was already mentioned earlier in the entry).

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
```diff
@@ -0,0 +1,72 @@
+## Syntax
+
+Given a  `string`, the `.endswith()` method can be used in the following way:
+
```
Was there a code snippet that was supposed to go here (using \```pseudo)? Something like this?

```pseudo
string.endswith(value, start, end)
```

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
```diff
@@ -0,0 +1,72 @@
+
+The `value` is required argument that is checked for whether it exists at the beginning of the `string`.  It is also case-sensitive.
+
+Conversely, the `value` can be tested whether it is the end of a particular substring of `string` by using the optional `start` and `end` index arguments.
```
```suggestion
Optionally, the `value` can be tested for whether it is at the end of a particular substring within the `string`. This is done within the `start` and `end` index arguments.
```

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
```diff
@@ -0,0 +1,72 @@
+## Examples
+
+The following checks for the value `g` at the end of the `example` string:
+
```
I think it would. be better to consolidate these snippets into two: 

* One code snippet for the Python. \```py
* One output snippet for the printed boolean values. \```shell
* After the snippets should be 2-3 sentences explaining how each `check_` gets its value.

```py
example_str = "This is a string"

check_A = example_str.endswith("g")
check_B = example_str.endswith("s")
check_C = example_str.endswith("st", 5, 12)

print("A: ", check_A)
print("B: ", check_B)
print("C: ", check_C)
```

And the output would be:

```shell
A: True
B: False
C: True
```

Dusch4593 replied:
Just looked at it and it's looking better! 👍 I left one more suggestion for the snippet below.

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
```diff
@@ -0,0 +1,72 @@
+
+Conversely, the `value` can be tested whether it is the end of a particular substring of `string` by using the optional `start` and `end` index arguments.
+
+## Examples
```
```suggestion
## Example
```

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
```diff
@@ -0,0 +1,72 @@
+
+## Syntax
+
+Given a  `string`, the `.endswith()` method can be used in the following way:
```
```suggestion
Given a `string`, the `.endswith()` method can be used in the following way:
```

Dusch4593 reviewed (changes requested):
@Pandz18 Hey there! Thanks for the changes 😄 I left some comments/suggestions for you to look through. This is shaping up to be a pretty solid entry! 🚀 🙏

yangc95 reviewed (commented):
@Pandz18 I have also gone ahead and left some minor comments/suggestions.

Please have a look on your available time so we can get this entry up 👌🏻

yangc95 commented on content/python/concepts/strings/terms/endswith/endswith.md:
```diff
@@ -0,0 +1,74 @@
+
+Given a `string`, the `.endswith()` method can be used in the following way:
+
+The `value` is a required argument that is checked if it exists at the end of the `string`.  It is also case-sensitive.
+
+Optionally, the `value` can be tested for whether it is at the end of a particular substring within the `string`. This is done within the `start` and `end` index arguments.
```
I think these read better as bullet points but I'm wondering about the general opinion on this @Dusch4593 @KTom101 @Pandz18 

If we decide to keep this in paragraph form, can we edit some of the wording?
```suggestion
The required `value` argument is checked if it exists at the end of the `string`.  It is also case-sensitive.

Optionally, the `value` can be tested for whether it is at the end of a particular substring within the `string` using the `start` and `end` index arguments.
```

KTom101 replied:
@yangc95 I think it can be condensed a little and changed into bullet points. Something like below maybe? 🤔

```suggestion
-`value` is a required, case-sensitive argument that is checked for at the end of the `string`.

- `value` can also be tested with the optional `start` and `end` index arguments to test whether it is at the end of a particular substring within the `string`.
```

But I think it could also be kept as a paragraph, reworded, and remove line 25 which makes it sound like a list is going to be followed after.

Pandz18 replied:
@yangc95 I would also think we can change it to a list. I can try to re-word it and will update it as soon as I can!

yangc95 replied:
Cool, thanks for the input!

I also think line 25 can be improved upon by mentioning something about the parameters being passed in rather than just "using" the method in certain ways. Alternatively (which I agree with) we can also remove that line as @KTom101 suggested.

If line 25 is removed, I think the rest can stay in paragraph format 👌🏻

Pandz18 replied:
> Cool, thanks for the input!
> 
> I also think line 25 can be improved upon by mentioning something about the parameters being passed in rather than just "using" the method in certain ways. Alternatively (which I agree with) we can also remove that line as @KTom101 suggested.
> 
> If line 25 is removed, I think the rest can stay in paragraph format 👌🏻

If it's fine with you, I can try to re-word it as a list and if you feel it would be better, leave it as is.

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
```diff
@@ -0,0 +1,62 @@
+print("A: ", check_A)
+print("B: ", check_B)
+print("C: ", check_C)
+
```
```suggestion
```
```

Dusch4593 replied:
Let's end the \```py code snippet here, add a new line, and then begin a new \```shell output snippet with the output lines (except for "# Output: ")

```
A: True
B: False
C: True
```

Dusch4593 commented on content/python/concepts/strings/terms/endswith/endswith.md:
````diff
@@ -0,0 +1,62 @@
+# Output C: True
+```
+
+From the above example, for ```check_A``` the output is True as the ```.endswith()``` function checks whether the the string ```example_str``` ends with the character "g".
````
```suggestion
From the above example, for `check_A` the output is True as the `.endswith()` function checks whether the the string `example_str` ends with the character "g".
```

When referencing a piece of code inline, let's surround with just one backtick on each side. Three backticks should be used only for code snippets that span multiple lines.

Pandz18 replied:
> When referencing a piece of code inline, let's surround with just one backtick on each side. Three backticks should be used only for code snippets that span multiple lines.

Understood, thank you!

Pandz18 commented:
> @Pandz18 Hey there! Thanks for the changes 😄 I left some comments/suggestions for you to look through. This is shaping up to be a pretty solid entry! 🚀 🙏

Thank you @Dusch4593 ! Will definitely go through them!

Pandz18 commented:
> @Pandz18 I have also gone ahead and left some minor comments/suggestions.
> 
> Please have a look on your available time so we can get this entry up 👌🏻

Hi @yangc95 , will go through them!

Pandz18 commented:
@yangc95 @Dusch4593 @KTom101   I've updated the file,  hope it's better now !

Dusch4593 reviewed (approved):
@Pandz18 Looks solid! 👍 

I think this is ready to merge, but I made a small edit in 9916117c422894f3081514b2baa2abaedc19ebee where I added a transition sentence in between the code snippets. Let's always make sure there is text between the snippets. 😄

yangc95 commented on content/python/concepts/strings/terms/endswith/endswith.md:
````diff
@@ -0,0 +1,65 @@
+C: True
+```
+
+From the above example, for `check_A` the output is True as the `.endswith()` function checks whether the the string `example_str` ends with the character "g".
````
```suggestion
From the above example, for `check_A` the output is `True` as the `.endswith()` function checks whether the `example_str` string ends with the character "g".
```

yangc95 commented on content/python/concepts/strings/terms/endswith/endswith.md:
````diff
@@ -0,0 +1,65 @@
+string.endswith(value, start, end)
+```
+
+Given a `string`, the `.endswith()` method can be used in the following way:
+
+- The required `value` argument, which is checked if it exists at the end of the `string`. It is also case-sensitive.
+
+- Optionally, the `value` can be tested for whether it is at the end of a particular substring within the `string` using the `start` and `end` index arguments.
````
Still thinking that we don't need those back ticks for string because we're referring to it as some value's data type rather than a required code component. Wondering about everyone's opinion on this @Dusch4593, @KTom101

Pandz18 replied:
> Still thinking that we don't need those back ticks for string because we're referring to it as some value's data type rather than a required code component. Wondering about everyone's opinion on this @Dusch4593, @KTom101

I agree that we do not need them, my bad for that will update the file.

Pandz18 replied:
@yangc95 is it fine now?

yangc95 replied:
Perfecto 👌🏻 

Also, it's all good 😊 For context, we made a similar change in another entry. Just making sure it is consistent.

yangc95 replied:
Oh shoot, but can we place those back on the value, start, and end keywords?

Pandz18 replied:
> Oh shoot, but can we place those back on the value, start, and end keywords? @Pandz18

Yeah sure, sorry bout that.

Pandz18 replied:
> Perfecto 👌🏻
> 
> Also, it's all good 😊 For context, we made a similar change in another entry. Just making sure it is consistent.

I understand 🙂

yangc95 reviewed (commented):
Caught some things! Also added something for discussion

Dusch4593 commented:
How is everyone feeling about this PR? Anything left to check before we merge?

cc: @Pandz18 @yangc95 @KTom101

yangc95 reviewed (approved):
Looks ready for merge!

Dusch4593 commented:
Congrats @Pandz18 for contributing another solid entry to Docs! 🚀 🎉 🎊 When the entry is live, it can be found here: https://www.codecademy.com/resources/docs/python/strings/endswith

Pandz18 commented:
> Congrats @Pandz18 for contributing another solid entry to Docs! 🚀 🎉 🎊 When the entry is live, it can be found here: https://www.codecademy.com/resources/docs/python/strings/endswith

Thank you so much @Dusch4593 @yangc95 @KTom101 !! 🚀 🎉

Dusch4593 commented:
@Pandz18 While I will always say that experience is the absolute best teacher, there are definitely some resources out there that I've found particularly helpful when learning to write good technical content/documentation:

- [Write the Docs](https://www.writethedocs.org/) - for learning about possible conferences/meetups and connect with others online through Twitter and Slack!
- [Google's courses on Tech Writing](https://developers.google.com/tech-writing/overview) are also pretty good, in my opinion. They really helped me with understanding how to better think like a tech writer and what to look for when reviewing/writing content.

yangc95 commented:
> If y'all wouldn't mind be asking, I want to become better at technical writing, what would y'all suggest that I do? Should I read/watch articles/courses/videos for guidance or learn from my contributions on here? 🙂

@Pandz18 That's awesome to hear!

+1 on @Dusch4593's recommendation. It's important to stay connected with the community. Also, I might check out those courses on technical writing from Google as well 👀 

💡 I'll first recommend Codecademy courses first because all the courses break down a topic into digestible chunks. If you're learning a particular language or tool with some goal in mind, try following the thought process of how each part of the lesson breaks down a certain point, and try to not leave any rocks unturned if you're unsure about the way something is explained. Try explaining something in your own words and write those down as notes if writing things down helps aid in your learning!

I really love tutorials as well, and when they're in documentation format like this one from Apple Developer (Intermediate/advanced) on [SwiftUI Essentials](https://developer.apple.com/tutorials/swiftui#swiftui-essentials), it's fun going through it ☺️

KTom101 commented:
@Pandz18 To add to what @yangc95 and @Dusch4593 have said, it's a good idea to have someone with no technical experience read your content. Pick out concepts that a beginner could understand and write about them in your own words. Share it with someone who has little to no knowledge in the subject and see if they can follow along. 

Also, from my experience, I found tutoring others helps with technical writing too (though it may seem unrelated). It really allows you to see what you can work on when explaining things. You can take note of people's reactions and figure out what works and what doesn't. You can even write out detailed lesson plans that you could use to tutor with. In the past, I've used lesson plans to explain concepts to myself first before explaining them to another person. 😄

AaronJohnson02 commented:
Hey, i apologise for the late reply, thank y'all so much for the advice. I will definitely try out the codecademy and Google courses whenever possible and also will definitely try to teach my peers to improve the skill.
